### PR TITLE
DO-3697: setting up pipeline for janna-api

### PR DESCRIPTION
**Pull request by therealpuneeth20**
Added Jenkins file.
fix lint issues.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        def buildNumber = env.BUILD_ID
+        def branch = env.BRANCH_NAME.replace('/', '-')
+
+        stage('BUILD'){
+            sh "make all IMAGE_TAG='${branch}'"
+        }
+
+        stage('DOCKER PUSH'){
+            sh "make push IMAGE_TAG='${branch}'"
```
Comment (vterdunov) [inline Jenkinsfile]: Please add push `:short_git_hash` (7 symbols) as well.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+    }
+} catch (e) {
+    manager.buildFailure()
+    echo e.toString()
+} finally {
+    Notification.send(this, currentBuild.currentResult == 'SUCCESS', ['#devops-pipeline'])
+}
```
Comment (vterdunov) [inline Jenkinsfile]: Need a final new line

File: Makefile [hunk 3]
```diff
@@ -25,14 +25,20 @@ OPENAPI_GENERATOR_CLI_VERSION = v3.2.2
...
 
+.PHONY: get
+
+get:
+	go get -u
+
 .PHONY: docker
 docker: ## Build Docker container
-	docker build --tag=${DOCKER_REGISTRY}$(IMAGE_NAME):$(COMMIT) --tag=${DOCKER_REGISTRY}$(IMAGE_NAME):latest --build-arg=GITHUB_TOKEN=${GITHUB_TOKEN} --file build/Dockerfile .
+	docker build -t ${DOCKER_REGISTRY}${IMAGE_NAME}:${IMAGE_TAG} --build-arg=GITHUB_TOKEN=${GITHUB_TOKEN} --file build/Dockerfile .
```
Comment (vterdunov) [inline Makefile]: Best practice is to keep long command names/flags in scripts. In more obviously.  So, please, use `--tag`.

File: Makefile [hunk 4]
```diff
@@ -25,14 +25,20 @@ OPENAPI_GENERATOR_CLI_VERSION = v3.2.2
...
 
 .PHONY: push
 push: ## Push docker container to registry
-	docker tag $(IMAGE_NAME):$(COMMIT) $(IMAGE_NAME):$(TAG)
-	docker push $(IMAGE_NAME):$(TAG)
+	#docker tag ${DOCKER_REGISTRY}$(IMAGE_NAME):$(COMMIT) ${DOCKER_REGISTRY}$(IMAGE_NAME):$(TAG)
```
Comment (vterdunov) [inline Makefile]: Please, remove those comments. Let's keep the codebase clean.

File: internal/endpoint/vm_disk_resize.go [hunk 5]
```diff
@@ -3,12 +3,13 @@ package endpoint
...
 	"context"
 	"errors"
-	"github.com/go-kit/kit/endpoint"
-	"github.com/virtualinstruments/janna-api/internal/service"
-	"github.com/virtualinstruments/janna-api/internal/types"
 	"log"
 	"regexp"
 	"strconv"
+
+	"github.com/go-kit/kit/endpoint"
```
Comment (vterdunov) [inline internal/endpoint/vm_disk_resize.go]: Ok for now, but in next time it would better to separate to another PR. It's not related to build pipeline.

File: Jenkinsfile (hunk 1 above)
Comment (therealpuneeth20) [reply Jenkinsfile]: Makefile will take care of push with git hash.
